core_lib/src/interface/editor.cpp

```diff
@@ -929,15 +929,25 @@ KeyFrame* Editor::addKeyFrame(int layerNumber, int frameIndex)
         return nullptr;
     }
 
+    // Find next available space for a keyframe (where either no key exists or there is an empty sound key)
     while (layer->keyExists(frameIndex))
     {
-        frameIndex += 1;
+        if (layer->type() == Layer::SOUND && static_cast<SoundClip*>(layer->getKeyFrameAt(frameIndex))->fileName().isEmpty()
+                && layer->removeKeyFrame(frameIndex))
+        {
+            break;
+        }
+        else
+        {
+            frameIndex += 1;
+        }
```

Comment:
I'm not sure it's a good idea to move to the next available keyframe implicitly, I'm thinking that we already have complains with the scrubber moving on its own.

Reply:
This isn't anything new, it already does this. You obviously can't add a new keyframe where one already exists, so it makes some sense to find the next available space to insert a keyframe. I think people would also complain about the add new frame action doing nothing if we did not do it this way.

Reply:
True, what I meant is that we had a popup that would appear when trying to create on the same frame for sound layer but I guess this is more consistent..